edgedb/protocol/codecs/codecs.pyx

```diff
@@ -510,6 +515,94 @@ cdef config_memory_decode(pgproto.CodecContext settings, FRBuffer *buf):
     return datatypes.ConfigMemory(bytes=bytes)
 
 
+DEF PGVECTOR_MAX_DIM = 16000
```

Comment:
What's special about this constant that we have to encode this on client?

I mean the datatypes allow encoding 32767 (or even 64k) of values, right? So it's just a matter of time when server (i.e. pgvector extension) might start allowing bigger values. Or is there any extra reserved bit or something?

Reply:
Good point

Reply:
> Vectors can have up to 16,000 dimensions.

From the pgvector docs. What will it do if I try to pass an array of 16001 floats? Crash? Trim? Submit and store 16001?

Reply:
Produces an error on the postgres side, which we turn into an InvalidValueError

Reply:
OK, sgtm